src/Microsoft.ApiDesignGuidelines.Analyzers/UnitTests/CollectionsShouldImplementGenericInterfaceTests.cs

```diff
@@ -12,12 +12,366 @@ public class CollectionsShouldImplementGenericInterfaceTests : DiagnosticAnalyze
     {
         protected override DiagnosticAnalyzer GetBasicDiagnosticAnalyzer()
         {
-            return new BasicCollectionsShouldImplementGenericInterfaceAnalyzer();
+            return new CollectionsShouldImplementGenericInterfaceAnalyzer();
         }
 
         protected override DiagnosticAnalyzer GetCSharpDiagnosticAnalyzer()
         {
-            return new CSharpCollectionsShouldImplementGenericInterfaceAnalyzer();
+            return new CollectionsShouldImplementGenericInterfaceAnalyzer();
+        }
+
+        [Fact]
+        public void Test_WithCollectionBase()
+        {
+            #region CSharp Test
```

Comment:
I'd omit the `#region` directives. I don't see them in any other analyzers I've looked at (and for that matter you didn't use them in any other test method).